pandas/_libs/missing.pyx

```diff
@@ -113,7 +118,18 @@ cpdef bint checknull(object val):
     The difference between `checknull` and `checknull_old` is that `checknull`
     does *not* consider INF or NEGINF to be NA.
     """
-    return val is C_NA or is_null_datetimelike(val, inat_is_null=False)
+    return (
+        val is C_NA
+        or is_null_datetimelike(val, inat_is_null=False)
+        or is_decimal_na(val)
```

Comment:
the list of what is considered null in docstring maybe could be updated.

also for consistency when using pandas.options.mode.use_inf_as_na, what about checknull_old?

Reply:
> the list of what is considered null in docstring maybe could be updated.

just added this to my next "collected misc" branch

> also for consistency when using pandas.options.mode.use_inf_as_na, what about checknull_old?

i guess you're referring to `Decimal("inf")`?  my inclination is to let that sleeping dog lie